kong/db/dao/plugins.lua

```diff
@@ -310,19 +310,35 @@ end
 
 
 function Plugins:select_by_cache_key(key)
-  -- try new format
+
+  -- first try new way
   local entity, new_err = self.super.select_by_cache_key(self, key)
-  if entity then
-    return entity
+
+  if not new_err then -- the step above didn't fail
+    -- we still need to check whether the migration is done,
+    -- because the new table may be only partially full
+    local schema_state = assert(self.db:schema_state())
+
+    -- if migration is complete, disable this translator function and return
+    if schema_state:is_migration_executed("core", "001_14_to_15") then
+      Plugins.select_by_cache_key = self.super.select_by_cache_key
+      return entity
+    end
   end
 
-  -- try old format
+  -- otherwise, we're either not started migrating, or we're migrating but
```

Comment:
grammar: `we're either not started migrating, or we're migrating but` -> `we either have not started migrating, or we're migrating but`